packages/controllers/src/plugins/PluginController.ts

```diff
@@ -431,21 +426,15 @@ export class PluginController extends EventEmitter {
       throw new Error('Expected array of plugin names.');
     }
 
-    const state = this.store.getState();
-    const newPlugins = { ...state.plugins };
-    const newPluginStates = { ...state.pluginStates };
-
-    pluginNames.forEach((name) => {
-      this._stopPlugin(name, false);
-      delete newPlugins[name];
-      delete newPluginStates[name];
+    this.update((state: any) => {
+      pluginNames.forEach((pluginName) => {
+        this._stopPlugin(pluginName, false);
+        delete state.plugins[pluginName];
+        delete state.pluginStates[pluginName];
+      });
     });
-    this._removeAllPermissionsFor(pluginNames);
 
-    this.updateState({
-      plugins: newPlugins,
-      pluginStates: newPluginStates,
-    });
+    this._removeAllPermissionsFor(pluginNames);
```

Comment:
I noticed that the order of this has changed now - it's after the state update rather than before. This seems fine though. I don't know of a reason this would matter.